### PR TITLE
Implemented api authentication via passport

### PR DESCRIPTION
**Pull request by dmitry-honcharuk**
Created passport authentication strategies;
Created routes for user authentication;

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -19,26 +20,55 @@ router.get('/', (req, res) => {
...
+                res.json({
+                    message: "User logged out",
+                    success: true
+                });
+            }
         });
+    }
+});
+
+router.get('/fail', (req, res) => {
```
Comment (evgeniyannenkov) [inline routes/api/users.js]: Maybe you could send status for all previous responses

Comment (dmitry-honcharuk) [inline routes/api/users.js]: Sorry didn't get it.
What all previous responses?

Comment (evgeniyannenkov) [inline routes/api/users.js]: check this file

Comment (dmitry-honcharuk) [inline routes/api/users.js]: '/fail' is a route for failureRedirect: '/api/users/fail', for passport middleware.
Here I have only 2 such routs, both have this redirect

File: routes/api/users.js [hunk 2]
```diff
@@ -19,26 +20,55 @@ router.get('/', (req, res) => {
...
-                data: user
-            });
+router.post('/login', passport.authenticate('login', {
+    failureRedirect: '/api/users/fail',
+    failureFlash: true
+}), (req, res) => {
+    res.json({success: true, user: req.user});
+});
+
+router.get('/current', (req, res) => {
```
Comment (evgeniyannenkov) [inline routes/api/users.js]: send status

File: routes/api/users.js [hunk 3]
```diff
@@ -19,26 +20,55 @@ router.get('/', (req, res) => {
...
         })
-        .catch(error => {
-            res.status(500).json({
-                success: false,
-                error
-            });
+    }
+});
+
+router.get('/logout', (req, res) => {
```
Comment (evgeniyannenkov) [inline routes/api/users.js]: send status

File: routes/api/users.js [hunk 4]
```diff
@@ -19,26 +20,55 @@ router.get('/', (req, res) => {
         });
 });
 
-router.post('/', (req, res) => {
-    const data = req.body;
-    const user = new User({
-        ...data,
-        registered: Date.now()
-    });
+router.post('/', passport.authenticate('register', {
```
Comment (evgeniyannenkov) [inline routes/api/users.js]: send status

Comment (dmitry-honcharuk) [inline routes/api/users.js]: 200 is default

File: routes/api/users.js (hunk 2 above)
Comment (dmitry-honcharuk) [inline routes/api/users.js]: 200 is default

File: routes/api/users.js (hunk 3 above)
Comment (dmitry-honcharuk) [inline routes/api/users.js]: 200 is default

**Pull request merged**